be/src/fs/fs.h

```diff
@@ -90,8 +90,8 @@ struct SequentialFileOptions {
 };
 
 struct RandomAccessFileOptions {
-    RandomAccessFileOptions() = default;
-
+    RandomAccessFileOptions(bool a) : skip_fill_local_cache(a){};
+    RandomAccessFileOptions() : skip_fill_local_cache(false){};
```

Comment:
Should remove all the no user-declared constructors to make it as an aggregate type to use designated initializers:
```C++

    RandomAccessFileOptions opts{.skip_fill_local_cache = _skip_fill_data_cache()};
```

Reply:
c++20 disallow
RandomAccessFileOptions() = default; and {} initialization list.

Reply:
https://godbolt.org/z/dEzdz7M5q

Reply:
Yes, `RandomAccessFileOptions() = default` is also a user-declared constructor. Therefore, when you remove it, it can use designated initializers.

https://godbolt.org/z/xrx1rj1bq

Reply:
According to [cppreference-aggregate_initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization):
> no user-**declared** or inherited constructors (since C++20)
> no user-**provided**, inherited, or explicit constructors | (since C++11)(until C++20)

Reply:
changed